pkg/controller/llmisvc/workload_tls_self_signed_istio.go

```diff
@@ -203,15 +217,45 @@ func (r *LLMInferenceServiceReconciler) expectedIstioDestinationRuleForWorkload(
 	if shadowSvc != nil {
 		dr.Spec.Host = network.GetServiceHostname(shadowSvc.GetName(), shadowSvc.GetNamespace())
 	}
-	if llmSvc.Spec.Router != nil && llmSvc.Spec.Router.Scheduler == nil {
-		dr.Spec.Host = network.GetServiceHostname(kmeta.ChildName(llmSvc.GetName(), "-kserve-workload-svc"), llmSvc.GetNamespace())
-	}
 
-	log.FromContext(ctx).V(2).Info("Expected destination rule for workload", "destinationrule", dr)
+	log.FromContext(ctx).V(2).Info("Expected destination rule for workload shadow service", "destinationrule", dr)
 
 	return dr, nil
 }
 
+func (r *LLMInferenceServiceReconciler) expectedIstioDestinationRuleForWorkload(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService) *istioapi.DestinationRule {
+	dr := &istioapi.DestinationRule{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kmeta.ChildName(llmSvc.GetName(), "-kserve-workload-svc"),
+			Namespace: llmSvc.GetNamespace(),
+			Labels: map[string]string{
+				"app.kubernetes.io/component": "llminferenceservice-workload",
+				"app.kubernetes.io/name":      llmSvc.GetName(),
+				"app.kubernetes.io/part-of":   "llminferenceservice",
+				"llm-d.ai/managed":            "true",
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(llmSvc, v1alpha1.LLMInferenceServiceGVK),
+			},
+		},
+		Spec: istionetworking.DestinationRule{
+			Host: network.GetServiceHostname(kmeta.ChildName(llmSvc.GetName(), "-kserve-workload-svc"), llmSvc.GetNamespace()),
+			TrafficPolicy: &istionetworking.TrafficPolicy{
+				Tls: &istionetworking.ClientTLSSettings{
+					Mode:               istionetworking.ClientTLSSettings_SIMPLE,
+					InsecureSkipVerify: &pbwrappers.BoolValue{Value: true},
```

Comment:
Is it ok to hardcode it?

Reply:
Yes, we don't really have a good solution to not have it yet